interface/ceed-operator.c

```diff
@@ -1180,6 +1211,543 @@ int CeedOperatorLinearAssembleAddPointBlockDiagonal(CeedOperator op,
   return 0;
 }
 
+/**
+   @brief Perform dense matrix-matrix multiply C += A B
+
+   Given m by n matrix A and n by p matrix B, add the matrix-matrix
+   product AB into the existing matrix C.
+
+   All matrices are stored in column-major order.
+
+   This could, probably should, be a GPU kernel.
+
+   @ref Utility
+**/
+static int MatrixMatrixMultiply(CeedInt m, CeedInt n, CeedInt p,
```

Comment:
See `CeedMatrixMultiply` in `ceed-basis.c`. I think if that function signature is added to `ceed-impl.h`, then we don't have to repeat this function here.

Reply:
I've done this, reducing code duplication (which is good) but somehow I'm not sure the function signature really belongs in `ceed-impl.h` which only contains objects and not functions. Let me know if there's a better place for it.

Reply:
Yeah, I think you're right. We could put it in `ceed-backend.h`. That has a lot of utility functions we don't want users accessing.

Reply:
So apparently it was already in `ceed-backend.h` but I didn't notice it...